apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -27,7 +27,9 @@ Services.obs.addObserver(function(document) {
 
     if (handlers) {
       handlers.forEach(function(handler) {
-        handler.call(null, parameters);
+        window.setTimeout(function() {
```

Comment:
Would be glad to know if you have ideas how to solve this better: in ```delete``` method we should resolve result promise when message is deleted _and_ fire event ```deleted``` if thread has been deleted as well. In _real_ mobileMessage API event always fired after result promise is resolved. So the only way I found that can simulate this behavior is timeout for the event and synchronous resolution of DOMRequest promise in delete method (```Services.DOMRequest.fireSuccess```). It's likely possible to make our mock as child class of some sort of built-in EventTarget that maybe has smth like ```dispatchEvent``` that can solve this issue out-of-the-box, but could not find how to do that :)